mercury/static/mercury/js/simulator.js

```diff
@@ -28,25 +30,172 @@ $(function () {
     function create_post() {
-        console.log("Entered create_post() function.");
+        console.log("Entered create_post() temp function.");
         $.ajax({
             url: "", // the endpoint
             type: "POST", // http method
             data: {
+                created_at: $('#post-created-at').val(),
                 temperature: $('#post-temperature').val(),
+
+            }, // data sent with the post request
+
+            // handle a successful response
+            success: function () {
+                generateValues();
+                console.log("POSTing was temp successful."); // another sanity check
+            },
+
+            // handle a non-successful response
+            error: function (xhr, errmsg, err) {
+                $('#results').html("<div class='alert-box alert radius' data-alert>Oops! We have encountered an error: " + errmsg +
+                    " <a href='#' class='close'>&times;</a></div>"); // add the error to the dom
+                console.log(xhr.status + ": " + xhr.responseText); // provide a bit more info about the error to the console
+            }
+        });
+    }
+
+    // Processes the form data and assigns the value to corresponding fields in the UI
+    function generateValues() {
+        let temperature = parseFloat($('#post-temperature').val());
+        $('#post-created-at').val(getDateTimenow());
+        $('#post-temperature').val(getNextValue(temperature,-5,5));
+    }
+
+    // This function returns current date time in the format "yyyy-mm-dd hh:min:ss"
+    function getDateTimenow(){
+        var now = new Date();
+        var yyyy = now.getFullYear();
+        var mm = ("0" + (now.getMonth() + 1)).slice(-2);
+        var dd = ("0" + now.getDate()).slice(-2);
+        var hours = ("0" + now.getHours()).slice(-2);
+        var minutes = ("0" + now.getMinutes()).slice(-2);
+        var seconds = ("0" + now.getSeconds()).slice(-2);
+        var curr_date = yyyy + '-' + mm + '-' + dd;
+        var curr_time = hours + ':' + minutes + ':' + seconds;
+        return curr_date + " " + curr_time;
+    }
+
+    //This function returns next sensor value, also makes sure that the value is in between 0 and 100
+    function getNextValue(sensorValue, min, max){
+        let MAX_VALUE = 100;
+        let MIN_VALUE = 0;
+        if((sensorValue-MIN_VALUE)<Math.abs(min)){
+            max = max-min;
+            min = 0;
+        }
+        else if((MAX_VALUE-sensorValue)<Math.abs(max)){
+            max = 0;
+        }
+        nextSensorValue = sensorValue + getRandomNumber(min,max);
+        return nextSensorValue;
+    }
+
+    //This function returns random number between min(inclusive) and max(exclusive)
+    function getRandomNumber(min,max){
+        let rand_num = Math.random()*(max-min)+min;
+        let rounded = rand_num.toFixed(2);
+        return parseFloat(rounded);
+    }
+
+
+    // This function gets cookie with a given name
+    function getCookie(name) {
```

Comment:
Same thoughts here; Why are we duplicating this function so many times in this code?

Reply:
Please note, there are lots of other functions duplicated in `simulator.js` that shouldn't be. There were too many to go through at once.